pkgs/applications/misc/gammu/default.nix

```diff
@@ -17,7 +17,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-aeaGHVxOMiXRU6RHws+oAnzdO9RY1jw/X/xuGfSt76I=";
   };
 
-  patches = [ ./bashcomp-dir.patch ./systemd.patch ];
+  patches = [ ./bashcomp-dir.patch ./systemd.patch ./gammu-config-dialog.patch ];
+  postPatch = ''
+    substituteInPlace utils/jadmaker --replace "/bin/bash" "${bash}/bin/bash"
+    substituteInPlace utils/gammu-config --replace "/bin/bash" "${bash}/bin/bash"
+    substituteInPlace utils/gammu-config --replace "NIX_PUT_DIALOG_PATH_HERE" "${dialog}/bin/dialog"
```

Comment:
We can use `substituteAll` on the patch itself. Maybe this is cleaner? For example:

```nix
{
  patches = [
    ./bashcomp-dir.patch
    ./systemd.patch
    (substituteAll {
      src = ./gammu-config-dialog.patch;
      dialog-path = "${dialog}/bin/dialog";
    })
  ];
}
```

where the patch would contain a line like `DIALOG=@dialog-path@`.

Reply:
We can also use `patchShebangs` for the other rules. See: https://discourse.nixos.org/t/what-is-the-patchshebangs-command-in-nix-build-expressions/12656#32-explicitly-6

Reply:
Using substituteAll on the patch is a good idea.

Reply:
No need to explicitly patch the shebangs. But `bash` has to be included in the `buildInputs` as `strictDeps = true;` is used in the derivation.